dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -156,6 +158,22 @@ public static void registerZenodoDOIForWorkflow(ApiClient zenodoClient, String d
 
         Deposit publishedDeposit = publishDepositOnZenodo(actionsApi, depositionID);
 
+        Object links = publishedDeposit.getLinks();
+        String conceptDoiUrl = (String)((LinkedHashMap)links).get("conceptdoi");
+        String conceptDoi = conceptDoiUrl;
+
+        // Remove the 'https://doi.org/' etc. prefix from the concept DOI
+        // e.g. https://doi.org/10.5072/zenodo.372767
+        try {
+            URI uri = new URI(conceptDoiUrl);
+            String[] segments = uri.getPath().split("/");
+            conceptDoi = segments[segments.length - 2] + "/" + segments[segments.length - 1];
+        } catch (URISyntaxException e) {
+            LOG.error("Could not extract workflow DOI. Error is " + e.getMessage(), e);
```

Comment:
This will continue after logging the error; don't you want to throw an exception here?

Reply:
We can just display the full concept DOI url if something goes wrong when attempting to display only the DOI part

Reply:
Do we do this for non-concept DOIs as well?

1. It feels odd that the format can vary so radically, e.g., the UI might always assume it's one format for display purposes, e.g., constructing a link, and then a totally different string appears.
2. Why do we strip it to begin with? Why not just store the url? Might be some reason I'm missing -- just asking.

Reply:
Zenodo considers a DOI to be of the form 10.5072/zenodo.1234. They call the url of the form https://doi.org/10.5072/zenodo.1234, which is returned in 'links' to be a Target URL...

![Screen Shot 2019-09-29 at 9 20 46 PM](https://user-images.githubusercontent.com/3650921/65849634-36c69580-e300-11e9-9aab-39bef3c0d715.png)